modules/supervisord.nix

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) 2019-2022, see AUTHORS. Licensed under MIT License, see LICENSE.
+
+{ pkgs, lib, config, ... }:
+let
+  inherit (lib) types;
+
+  cfg = config.supervisord;
+
+  format = pkgs.formats.ini {};
+
+  programType = types.submodule ({ name, config, ... }: {
+    options = {
+      enable = lib.mkOption {
+        description = lib.mdDoc ''
+          Whether to enable this program.
+        '';
+        type = types.bool;
+        default = true;
+      };
+      command = lib.mkOption {
+        description = lib.mdDoc ''
+          The command that will be run as the service's main process.
+        '';
+        type = types.str;
+      };
+      script = lib.mkOption {
+        description = lib.mdDoc ''
+          Shell commands executed as the service's main process.
+        '';
+        type = types.lines;
+        default = "";
+      };
+      path = lib.mkOption {
+        description = lib.mdDoc ''
+          Packages added to the service's PATH environment variable.
+        '';
+        type = types.listOf (types.either types.package types.str);
+        default = [];
+      };
+      autoRestart = lib.mkOption {
+        description = lib.mdDoc ''
+          Whether to automatically restart the process if it exits.
+
+          If `unexpected`, the process will be restarted if it exits
+          with an exit code not listed in the programs's `exitcodes`
+          configuration.
+        '';
+        type = types.either types.bool (types.enum [ "false" "true" "unexpected" ]);
+        default = "unexpected";
+      };
+      environment = lib.mkOption {
+        description = lib.mdDoc ''
+          Environment variables passed to the service's process.
+        '';
+        type = types.attrsOf types.str;
+        default = {};
+      };
+      extraConfig = lib.mkOption {
+        description = lib.mdDoc ''
+          Extra structured configurations to add to the [program:x] section.
+        '';
+        type = types.attrsOf types.str;
```

Comment:
Because `renderAtom` supports also boolean, it would be reasonable to support booleans here too

Reply:
Ok, added